var/spack/repos/builtin/packages/tcl/package.py

```diff
@@ -93,6 +93,17 @@ def command(self):
             os.path.realpath(self.prefix.bin.join("tclsh{0}".format(self.version.up_to(2))))
         )
 
+    def _find_script_dir(self):
```

Comment:
What about error handling? For libraries / headers an exception is thrown iirc.

Reply:
Overall think I should raise an exception here, as you say (that would generally preserve the current behavior although more explicitly). Regarding the behavior of `.libs`/`.headers`:

* `find` (and associated implementations like `find_libraries`) return `[]` and don't fail.
* The default implementation of `.libs` does fail, although I don't know if those semantics need to be copied here on that account (i.e. the user could decide just not to set the associated env variable)
* Without this PR, a failure to locate any files would fail when the code attempts to dereference the first element of an empty list

Reply:
I now raise an exception in `_find_script_dir` if nothing is found (which restores the prior behavior, but with a clearer error message).